java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -239,12 +233,52 @@ void setTrueBlock(Block trueBlock) {
     }
   }
 
+  private static class BytecodeCFGClassVisitor extends ClassVisitor {
+
+    private final BytecodeCFGMethodVisitor methodVisitor;
+    private final String methodSignature;
+    private boolean isFinalClass = false;
+
+    public BytecodeCFGClassVisitor(BytecodeCFGMethodVisitor methodVisitor, String targetedMethodSignatures) {
+      super(Opcodes.ASM5);
+      this.methodVisitor = methodVisitor;
+      this.methodSignature = targetedMethodSignatures;
+    }
+
+    @Override
+    public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+      isFinalClass = Flags.isFlagged(Flags.filterAccessBytecodeFlags(access), Flags.FINAL);
+      super.visit(version, access, name, signature, superName, interfaces);
+    }
+
+    @Override
+    public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+      if (name.equals(methodSignature.substring(methodSignature.indexOf('#') + 1, methodSignature.indexOf('(')))
+        && desc.equals(methodSignature.substring(methodSignature.indexOf('(')))) {
+        int methodFlags = Flags.filterAccessBytecodeFlags(access);
+        methodVisitor.isStaticMethod = Flags.isFlagged(methodFlags, Flags.STATIC);
+        methodVisitor.isVarArgs = Flags.isFlagged(methodFlags, Flags.VARARGS);
+        methodVisitor.isNonOverrideableMethod = methodCanNotBeOverriden(methodFlags);
+        return new JSRInlinerAdapter(methodVisitor, access, name, desc, signature, exceptions);
+      }
+      return null;
+    }
+
+    private boolean methodCanNotBeOverriden(int methodFlags) {
+      if (Flags.isFlagged(methodFlags, Flags.NATIVE)) {
+        return false;
+      }
+      return Flags.isNotFlagged(methodFlags, Flags.ABSTRACT) && (isFinalClass || Flags.isFlagged(methodFlags, Flags.PRIVATE | Flags.FINAL | Flags.STATIC));
+    }
+  }
+
   private static class BytecodeCFGMethodVisitor extends MethodVisitor {
     Map<Label, Block> blockByLabel = new HashMap<>();
     private Block currentBlock;
     private BytecodeCFG cfg;
     private boolean isStaticMethod;
     private boolean isVarArgs;
+    private boolean isNonOverrideableMethod;
```

Comment:
I would go for isOverrideable... double negation is hard. And in fact that would be : isOverrideableOrNative